packages/gatsby-link/src/index.js

```diff
@@ -4,6 +4,12 @@ import { Link } from "@reach/router"
 
 import { parsePath } from "./parse-path"
 
+if (!module.parent.id.includes(`gatsby-browser-entry`)) {
```

Comment:
module.parent doesn't work with webpack and even if it did, we're expecting the `caller` where `parent` is set to the first module that imported the current module (and therefore resulted it in being added to the resolve cache)